src/Swarm/Game/Step.hs

```diff
@@ -1484,6 +1494,18 @@ execConst c vs s k = do
     Concat -> case vs of
       [VText v1, VText v2] -> return $ Out (VText (v1 <> v2)) s k
       _ -> badConst
+    Char -> case vs of
+      [VInt i, VText t]
+        | i < 0 || i >= fromIntegral (T.length t) ->
+          raise Char ["Index", prettyValue (VInt i), "out of bounds for length", from @String $ show (T.length t)]
+        | otherwise -> return $ Out (VInt . fromIntegral . ord . T.index t . fromIntegral $ i) s k
+      _ -> badConst
+    MkText -> case vs of
+      [VInt n, f]
+        | n < 0 -> raise MkText ["Length must be nonnegative:", prettyValue (VInt n)]
+        | n == 0 -> return $ Out (VText "") s k
+        | otherwise -> return $ Out (VInt (n - 1)) s (FApp f : FMkText f (n - 1) [] : k)
```

Comment:
I am not sure I understand what is going on here. Is this supposed to be more efficient than using `++` from Swarm?

Reply:
Yes, it is more efficient, but I agree at the very least I would need to write some comments explaining how this works.  Though as I mentioned elsewhere I am now thinking of getting rid of this for the moment --- though we will probably need something similar to implement the build function for arrays (#98).